thirdparty/idstore/idstore.go

```diff
@@ -0,0 +1,108 @@
+package idstore
+
+import (
+	"context"
+	"fmt"
+
+	cide "github.com/ipfs/go-ipfs/thirdparty/cidextra"
+
+	mh "gx/ipfs/QmZyZDi491cCNTLfAhwcaDii2Kg4pwKRkhqQzURGDvY6ua/go-multihash"
+	bls "gx/ipfs/QmaG4DZ4JaqEfvPWt5nPPgoTzhc1tr1T3f4Nu9Jpdm8ymY/go-ipfs-blockstore"
+	cid "gx/ipfs/QmcZfnkapfECQGcLZaf9B79NRg7cRa9EnZh4LSbkCzwNvY/go-cid"
+	blocks "gx/ipfs/Qmej7nf81hi2x2tvjRBF3mcp74sQyuDH4VMYDGd1YtXjb2/go-block-format"
+)
+
+// idstore wraps a BlockStore to add support for identity hashes
+type idstore struct {
+	bs bls.Blockstore
+}
+
+func IdStore(bs bls.Blockstore) bls.Blockstore {
+	return &idstore{bs}
+}
+
+func extractContents(k *cid.Cid) (bool, []byte, error) {
```

Comment:
We'll probably want to move this (and maybe a few other things) to, e.g., `go-block`. We'll want this stuff usable outside of go-ipfs.

Reply:
See my earlier comment, I want to first worry about getting something worry and then we can move the my two new thirdparty packages to a better location.